src/pages/signin/PasswordForm.js

```diff
@@ -35,28 +42,54 @@ const propTypes = {
 };
 
 const defaultProps = {
-    account: {},
+    account: {
+        password: '',
+        twoFactorAuthCode: '',
+    },
 };
 
 class PasswordForm extends React.Component {
     constructor(props) {
         super(props);
 
+        this.changePassword = this.changePassword.bind(this);
+        this.changeTwoFactorAuthCode = this.changeTwoFactorAuthCode.bind(this);
         this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+        this.password = props.account.password;
+        this.twoFactorAuthCode = props.account.twoFactorAuthCode;
+
 
         this.state = {
             formError: false,
-            password: '',
-            twoFactorAuthCode: '',
         };
     }
 
+    /**
+     * Update the value of password in Onyx
+     *
+     * @param {String} newPassword
+     */
+    changePassword(newPassword) {
+        this.password = newPassword;
+        Onyx.merge(ONYXKEYS.ACCOUNT, {password: newPassword});
+    }
+
+    /**
+     * Update the value of twoFactorAuthCode in Onyx
+     *
+     * @param {String} newTwoFactorAuthCode
+     */
+    changeTwoFactorAuthCode(newTwoFactorAuthCode) {
+        this.TwoFactorAuth = newTwoFactorAuthCode;
+        Onyx.merge(ONYXKEYS.ACCOUNT, {twoFactorAuthCode: newTwoFactorAuthCode});
+    }
+
     /**
      * Check that all the form fields are valid, then trigger the submit callback
      */
     validateAndSubmitForm() {
-        if (!this.state.password.trim()
```

Comment:
I think we'll want to make sure we remove the password and 2fa code from `ONYXKEYS.ACCOUNT` in this function, since we don't want store those values in plaintext once the user has attempted to log in

Reply:
Do we want to do this here or inside `componentWillUnmount`

```js
    componentWillUnmount() {
        Onyx.merge(ONYXKEYS.ACCOUNT, {password: '', twoFactorAuthCode: ''});
    }
```

Reply:
Yep, that's a better location